### PR TITLE
Drones

### PR DESCRIPTION
**Pull request by ronanhub**
drones, place em, fly em, dominate the weak:

- Drones can be placed with the drone prefab
- Drones can be stationary if you have no patrol points
- Drones can move along a path of patrol points

### CODE REVIEW

File: WobblyStealthGame/Assets/Scripts/Enemies/PathMovement.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    [SerializeField] float moveSpeed;
+
+    void FixedUpdate()
+    {
+        if (PatrolPoints.Length > 0)
+        {
+            Vector3 directionVector = PatrolPoints[patrolPoint].position - transform.position;
+            if (directionVector.magnitude < patrolPointReachedThreshold)
+            {
+                patrolPoint = (patrolPoint+1) % PatrolPoints.Length;
```
Comment (MatthijsHofele) [inline WobblyStealthGame/Assets/Scripts/Enemies/PathMovement.cs:19]:
```suggestion
                patrolPoint = (patrolPoint + 1) % PatrolPoints.Length;
```

File: WobblyStealthGame/Assets/Scripts/Enemies/PathMovement.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            }
+            else
+            {
+                transform.position += directionVector.normalized*moveSpeed;
```
Comment (MatthijsHofele) [inline WobblyStealthGame/Assets/Scripts/Enemies/PathMovement.cs:23]:
```suggestion
                transform.position += directionVector.normalized * moveSpeed;
```

**Review by MatthijsHofele — CHANGES_REQUESTED** [review]

**Comment by MatthijsHofele** [comment]
Dronan

File: WobblyStealthGame/Assets/Scripts/Enemies/CCTVConeBehaviour.cs [hunk 3]
```diff
@@ -12,6 +12,7 @@ public class CCTVConeBehaviour : MonoBehaviour
     [SerializeField, Range(0f, 10f)]private float lowerDetectionAmount;
     [SerializeField, Range(0f, 100f)]private float detectonFalloffStartDistance;
     [SerializeField, Range(0f, 100f)]private float detectonFalloffRange;
+    [SerializeField] private Transform linecastStartTransform;
     [SerializeField]private AudioSource alarmSound;
```
Comment (Jarivanoort) [inline WobblyStealthGame/Assets/Scripts/Enemies/CCTVConeBehaviour.cs:16]:
```suggestion
    [SerializeField] private Transform linecastStartTransform;
    [SerializeField] private AudioSource alarmSound;
```

**Review by Jarivanoort — CHANGES_REQUESTED** [review]
Drone doesnt have collider

**Review by Jarivanoort — APPROVED** [review]

**Pull request merged**